flink-libraries/flink-table/src/test/scala/org/apache/flink/table/expressions/ScalarFunctionsTest.scala

```diff
@@ -352,6 +352,14 @@ class ScalarFunctionsTest extends ScalarTypesTestBase {
       "Flink~~~~xx")
   }
 
+  @Test
+  def testBin(): Unit = {
+    testSqlApi("BIN(12)", "1100")
+    testSqlApi("BIN(10)", "1010")
+    testSqlApi("BIN(0)", "0")
+    testSqlApi("BIN(f32)","1111111111111111111111111111111111111111111111111111111111111111")
+  }
+
```

Comment:
Would be better to add validation tests to `ScalarFunctionsValidationTest` to check the expected exception for unsupported operand types.